### PR TITLE
Jurisdictional Refactor Feature Merge

### PR DESCRIPTION
**Pull request by ColinBuyck**
# Pull Request Template

## Issue Overview

This PR addresses #319 

- [x] This change addresses the issue in full
- [ ] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

This PR merges the jurisdictional refactor feature branch into main. it also includes 2 minor testing fixes that came to light in this PR.

## How Can This Be Tested/Reviewed?

The best way to test this is to mimic how this will work when releasing this to prod:

1. **Before you pull down this change** - in the main branch do a reseed
2. Pull this change down and in the backend/core directory run `yarn db:migration:run`
3. Migration should run correctly and should be able to start up the app
4. The partner site should show the following changes:
  - When adding a new partner user all of the new jurisdictions should exist
  - When adding a new listing the jurisdiction dropdown should appear with all of the new jurisdictions as options
  - AMI charts should be tied to appropriate jurisdictions

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have added QA notes to the issue with applicable URLs
- [ ] I have performed a self-review of my own code
- [ ] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [ ] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

File: backend/core/src/migration/1695143897902-add-jurisdictions.ts [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+      } else if (user.is_partner) {
+        const userListings = await queryRunner.query(
+          `SELECT listings_id from listings_leasing_agents_user_accounts where user_accounts_id = '${user.user_id}'`
+        )
+        userListings.forEach(async (listing) => {
+          const matchingListing = existingListings.find(
+            (list) => list.id === listing.listings_id
+          )
+          const matchingJuris = existingJurisdictions.find((juris) => juris.name === matchingListing.county).id
+          console.log("matchingListing", matchingListing)
```
Comment (ColinBuyck) [inline backend/core/src/migration/1695143897902-add-jurisdictions.ts:79]: Assuming we want to delete these @ludtkemorgan

**Review by ColinBuyck — COMMENTED** [review]
LGTM 🟢 Just one comment. Can't fully approve though since I opened the PR @ludtkemorgan

File: backend/core/src/migration/1695143897902-add-jurisdictions.ts (hunk 1 above)
Comment (ludtkemorgan) [reply backend/core/src/migration/1695143897902-add-jurisdictions.ts]: good catch. If you are ok with it. I'll approve it for you. Most of the code was done by you and I reviewed that portion

Comment (ColinBuyck) [reply backend/core/src/migration/1695143897902-add-jurisdictions.ts]: Sounds good!

**Review by ludtkemorgan — APPROVED** [review]

**Pull request merged**